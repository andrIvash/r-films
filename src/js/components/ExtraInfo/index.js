// @flow
import React from 'react';
import helpers from '../../helpers';
import type { View } from '../../flow-types.js';

type Props = {
  data: string,
  view: View
}

function showInfo( props: Props ) {
  const { view, data } = props;
  return view === helpers.views.POSTER ?
    `Films by ${data} genre` :
    `${data} movies found`;
}

const ExtraInfo = (props: Props) => (
  <div className='extra-info'>
    <div className='extra-info__content'>
      {showInfo(props)}
    </div>
  </div>
);

export default ExtraInfo;
