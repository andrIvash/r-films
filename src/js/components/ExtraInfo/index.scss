.extra-info__content {
  font-family: $sign-font;
  font-size: rem(11px);
  font-weight: 700;
  color: $color-background;
}