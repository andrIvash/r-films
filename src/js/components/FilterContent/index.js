// @flow
import React from 'react';

type Props = {
 onFilterSelect: (ev: SynteticInputIvent) => void,
 selected: string
}

const FilterContent = (props: Props) => (
  <div className='filter-content'>
    <div className='filter-content__title'>Sort By</div>
    <div className='filter-content__group'>
      <label
        className={props.selected === 'release' ?
        'filter-content__label active' :
        'filter-content__label'}
        htmlFor='release'
        >
        <input
          autoComplete='off'
          className='filter-content__input'
          id='release'
          name='filter'
          onChange={(ev) => props.onFilterSelect(ev.target.value)}
          type='radio'
          value='release'
        /> release date
      </label>
      <label
        className={props.selected === 'rating' ?
        'filter-content__label active' :
        'filter-content__label'}
        htmlFor='rating'
        >
        <input
          autoComplete='off'
          className='filter-content__input'
          id='rating'
          name='filter'
          onChange={(ev) => props.onFilterSelect(ev.target.value)}
          type='radio'
          value='rating'
        /> rating
      </label>
    </div>
  </div>
);

export default FilterContent;
