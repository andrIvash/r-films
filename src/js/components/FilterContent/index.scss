.filter-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-family: $sign-font;
  font-size: rem(16px);
  font-weight: 500;
  color: $color-background;
}

.filter-content__label {
  margin-bottom: 0;
  margin-left: rem(15px);
  cursor: pointer;
}

.filter-content__label.active {
  color: $color-logo;
}

.filter-content__input {
  display: none;
}