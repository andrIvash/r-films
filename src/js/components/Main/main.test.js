import React from 'react';
import configureMockStore from 'redux-mock-store';
import CombinedMain, { Main } from './index';
import FilterContent from '../FilterContent';
import ExtraInfo from '../ExtraInfo';

const mockStore = configureMockStore();

let props = {};

describe('Main', () => {
  beforeEach(() => {
    props = {
      films: [
        {id: 1, title: 'title', release_date: '1992', genres: ['Drama']},
        {id: 2, title: 'title2', release_date: '1994', genres: ['Drama']},
      ],
      onFilterSelect: jest.fn(),
      view: 'COMMON',
    };
  });
  it('should be defined', () => {
    expect(Main).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = render(
      <Main {...props} /> );
    expect(wrapper).toMatchSnapshot();
  });
  it('should show an extra info ', () => {

    const wrapper = shallow(
      <Main {...props} /> );
    expect(wrapper.find(ExtraInfo)).toHaveLength(1);
    expect(wrapper.find(FilterContent)).toHaveLength(1);
  });
  it('should not show an extra info if no data', () => {
    props.films = [];

    const wrapper = shallow(
      <Main {...props} /> );
    expect(wrapper.find(ExtraInfo)).not.toHaveLength(1);
  });
  it('should change filter when onFilterSelect emit', () => {
    const wrapper = mount(
      <Main {...props} /> );
    wrapper.find(FilterContent).find('#release').simulate('change');
    expect(props.onFilterSelect).toHaveBeenCalledTimes(1);
    expect(props.onFilterSelect.mock.calls[0][0])
      .toEqual('release');
  });
});

describe('CombinedMain', () => {
    let wrapper, store;

    beforeEach(() => {
      const initialState = {
        selectDataFilter: {
          selected: 'rating',
        },
        filterData: {
          filteredFilms: [],
        },
      };
      store = mockStore(initialState);
      wrapper = shallow(
        <CombinedMain store={store} />,
      );
    });

    it('should show previously state value', () => {
      expect(wrapper.props().selected).toBe('rating');
    });

});
