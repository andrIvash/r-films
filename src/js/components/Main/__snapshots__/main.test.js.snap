// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
<main
  class="content app__main"
>
  <div
    class="content__top"
  >
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div
          class="extra-info"
        >
          <div
            class="extra-info__content"
          >
            2 movies found
          </div>
        </div>
        <div
          class="filter-content"
        >
          <div
            class="filter-content__title"
          >
            Sort By
          </div>
          <div
            class="filter-content__group"
          >
            <label
              class="filter-content__label"
              for="release"
            >
              <input
                autocomplete="off"
                class="filter-content__input"
                id="release"
                name="filter"
                type="radio"
                value="release"
              />
               release date
            </label>
            <label
              class="filter-content__label active"
              for="rating"
            >
              <input
                autocomplete="off"
                class="filter-content__input"
                id="rating"
                name="filter"
                type="radio"
                value="rating"
              />
               rating
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="container"
  >
    <div
      class="content-info"
    >
      <ul
        class="film-list"
      >
        <li
          class="film-list__item"
        >
          <div
            class="film-item"
          >
            <div
              class="film-item__img-wrap"
            >
              <img
                alt="film"
                class="film-item__img"
              />
            </div>
            <div
              class="film-item__info"
            >
              <div
                class="film-item__title"
              >
                title
              </div>
              <div
                class="film-item__year"
              >
                1992
              </div>
            </div>
            <div
              class="film-item__genre"
            >
              Drama
            </div>
          </div>
        </li>
        <li
          class="film-list__item"
        >
          <div
            class="film-item"
          >
            <div
              class="film-item__img-wrap"
            >
              <img
                alt="film"
                class="film-item__img"
              />
            </div>
            <div
              class="film-item__info"
            >
              <div
                class="film-item__title"
              >
                title2
              </div>
              <div
                class="film-item__year"
              >
                1994
              </div>
            </div>
            <div
              class="film-item__genre"
            >
              Drama
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</main>
`;
